hack/generator/pkg/astmodel/kubebuilder_validations.go

```diff
@@ -130,7 +131,7 @@ func ValidateMaxLength(length int64) KubeBuilderValidation {
 	return KubeBuilderValidation{MaxLengthValidationName, length}
 }
 
-func ValidatePattern(pattern regexp.Regexp) KubeBuilderValidation {
+func ValidatePattern(pattern *regexp.Regexp) KubeBuilderValidation {
```

Comment:
The current name `ValidatePattern` reads as though it's doing the validation, not creating a validation item using a pattern.

Suggest improving the name, maybe `AsValidationPattern()`?